chargeio/infrastructure/Requestor.cs

```diff
@@ -0,0 +1,117 @@
+                }
+            }
+        }
+
+        private static string Put(string url, string putData, string contentType, string secretKey)
+        {
+            using(HttpClient client = new HttpClient()) {
+                try {
+                    var byteArray = Encoding.ASCII.GetBytes(Configuration.SecretKey + ":");
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    client.Timeout = TimeSpan.FromMilliseconds(Configuration.HttpTimeout);
+                    client.DefaultRequestHeaders.Add("User-Agent", GetUserAgent());
+                    var result = client.PutAsync(url, new StringContent(putData, Encoding.UTF8, contentType));
+                    return result.Result.Content.ReadAsStringAsync().Result;
+                } catch (Exception e) {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+        }
+
+        private static string Post(string url, string putData, string contentType, string secretKey)
+        {
+            using(HttpClient client = new HttpClient()) {
+                try {
+                    var byteArray = Encoding.ASCII.GetBytes(Configuration.SecretKey + ":");
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    client.Timeout = TimeSpan.FromMilliseconds(Configuration.HttpTimeout);
+                    client.DefaultRequestHeaders.Add("User-Agent", GetUserAgent());
+                    var result = client.PostAsync(url, new StringContent(putData, Encoding.UTF8, contentType));
+                    return result.Result.Content.ReadAsStringAsync().Result;
+                } catch (Exception e) {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+        }
+
+        private static string Delete(string url)
+        {
+            using(HttpClient client = new HttpClient()) {
+                try {
+                    var byteArray = Encoding.ASCII.GetBytes(Configuration.SecretKey + ":");
+                    client.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Basic", Convert.ToBase64String(byteArray));
+                    client.Timeout = TimeSpan.FromMilliseconds(Configuration.HttpTimeout);
+                    client.DefaultRequestHeaders.Add("User-Agent", GetUserAgent());
+                    var result = client.DeleteAsync(url);
+                    return result.Result.Content.ReadAsStringAsync().Result;
+                } catch (Exception e) {
+                    Console.WriteLine(e);
+                    return null;
+                }
+            }
+        }
+
+        private static string GetUserAgent()
+        {
+            return "ChargeIO .net Client";

```

Comment:
Why was the package version dropped from the user agent? We use this to track and distinguish library versions being used to access the system.